### PR TITLE
Idml validation

### PR DESCRIPTION
**Pull request by vibby**
First step in idml validation.

### CODE REVIEW

File: application/views/twigs/products.twig [hunk 1]
```diff
@@ -158,169 +161,8 @@
...
-					</div>
-					<div class="modal-footer">
-						<button role="button" class="btn btn-primary" type="submit">
-							Create Product
-						</button>
-					</div>
-				</div>
-			</div>
-		</form>
+		{% include 'twigs/modals/_product_add.twig' %}
```
Comment (davrv93) [inline application/views/twigs/products.twig]: Thanks  @vibby , it remids me about ng-template on angular :)  I like this way of including html

File: public/assets/js/app.js [hunk 2]
```diff
@@ -74,8 +74,18 @@ function ajax( url, data, $btn, $response_region, success_callback ) {
...
+			var errorMessage = '';
+			if (jqXHR.status == 500) {
+			  errorMessage='Server side 500 error';
+			} else if (jqXHR.status == 404) {
+			  errorMessage='Not found 404 error';
+			} else if (jqXHR.status == 404) {
+			  errorMessage='Not allowed 403 error. Are you still logged ?';
+			} else {
+			  errorMessage='An error occured';
+			}
```
Comment (davrv93) [inline public/assets/js/app.js]: Good Try / Catch handling , Thanks!

File: src/Idml/Entity/Holder.php [hunk 3]
```diff
@@ -104,11 +89,22 @@ public function getStory($storyKey): Story
...
+        return $this->stories;
+    }
+    
     public function getSections()
     {
         if (!$this->sections) {
-            foreach ($this->getPackage()->getStories() as $storyKey => $storyNode) {
-                $story = new Story($storyKey, $storyNode, StoryBasedOnTags::class);
+            foreach ($this->getStories() as $story) {
                 $this->sections = array_merge($this->sections, $story->getSections());
```
Comment (davrv93) [inline src/Idml/Entity/Holder.php]: These two lines seems so really much simpler and efficent than the way that we were using for detecting it, I have follow the traceability around the functions, Thanks, good contribution

Comment (vibby) [reply src/Idml/Entity/Holder.php]: I also kept the sections separation based on separator elements, but it is not used. For later, we could imagine that the user select the way content is separated when he uploads the imdl.

**Pull request merged**